src/StationAPIConnector.php

```diff
@@ -0,0 +1,249 @@
+      ];
+    }
+    catch (\Exception $e) {
+      $this->logger->get('tide_station_locator connect API.')
+        ->error('BadResponse error: @error', ['@error' => $e->getResponse()]);
+      return FALSE;
+    }
+  }
+
+  /**
+   * Get API response.
+   *
+   * @param string $client
+   *   Client ID.
+   * @param string $accessToken
+   *   Access token.
+   * @param string $rdm_auth
+   *   Rdm Auth.
+   * @param bool $delta
+   *   Delta boolean.
+   *
+   * @return mixed
+   *   Full data.
+   */
+  public function get_api_response($client, $accessToken, $rdm_auth, $delta) {
+    try {
+      // Make an API request.
+      $response = $client->post('technology/rdm/3.0.0/STATION_LOCATION/search', [
+        RequestOptions::HEADERS => [
+          'Authorization' => 'Bearer ' . $accessToken,
+          'rdmAuthorization' => $rdm_auth,
+          'Accept' => '*/*',
+          'Content-Type' => 'application/json',
+          'Accept-Encoding' => 'gzip, deflate, br',
+        ],
+        RequestOptions::JSON => $this->api_query_payload($delta),
+      ]);
+
+      if ($response->getStatusCode() == '200') {
+        // Get the config.
+        $config = $this->configFactory->getEditable('tide_station_locator.settings');
+        // Update 'last_api_access'.
+        $config->set('last_api_access', date('Y-m-d h:i:s'))->save();
+
+        // Get the response body.
+        $body = $response->getBody()->getContents();
+
+        $full_data = json_decode($body, TRUE);
+
+        return $full_data;
+      }
+    }
+    catch (BadResponseException $e) {
+      $this->logger->get('tide_station_locator API response')
+        ->error('BadResponse error: @error', ['@error' => $e->getResponse()]);
+      return FALSE;
+    }
+
+    return [];
+  }
+
+  /**
+   * Helper method to get the Query Payload.
+   *
+   * @return array
+   *   Payload.
+   */
+  public function api_query_payload($delta = FALSE) {
+    // Get the 'last_api_access'.
+    $config = $this->configFactory->getEditable('tide_station_locator.settings');
+
+    if ($delta && !empty($config->get('last_api_access'))) {
+      $payload = [
+        'name' => 'STATION_LOCATION',
+        "productKey" => -1,
+        'attributes' => [
+          [
+            'name' => 'RECORD_MODDATE',
+            'value' => [$config->get('last_api_access')],
+            'operator' => 'gt',
+            'caseSensitive' => FALSE,
+          ],
+        ]
+      ];
+    }
+    else {
+      $payload = [
+        'name' => 'STATION_LOCATION',
+        'attributes' => [
+          [
+            'name' => 'ACTIVE',
+            'value' => ['Y'],
+            'caseSensitive' => FALSE,
+          ],
+        ],
+        'startIndex' => 1,
+        'recordCount' => 5,
+      ];
+    }
+
+    return $payload;
+  }
+
+  /**
+   * Helper method to get the 'last_access_api'.
+   *
+   * @return string
+   *   Config value last_api_access.
+   */
+  public function get_last_access_api() {
```

Comment:
Where are we using this? why are we setting the value to now if no value is found?